### PR TITLE
Implement adjacency functions (rook and queen) and creation of R-Tree

### PR DESCRIPTION
**Pull request by matthewkol186**
(Depends on #33.)

Pretty small PR - just implementing a queen/rook adjacency function and also a function that constructs an R-Tree, given an array of minimum bounding rectangles.

### CODE REVIEW

File: src/graph.jl [hunk 1]
```diff
@@ -39,6 +39,35 @@ function min_bounding_rect(coords::Vector{Vector{Vector{Vector{Float64}}}})::Tup
...
 
 
+function create_rtree(minimum_bounding_rects::Array{Tuple{Array{Float64,1},Array{Float64,1}},1})::LibSpatialIndex.RTree
+    """ Given an array of minimum bounding rectangles, constructs an R-Tree
+        that will make it easier to identify candidates for intersecting nodes.
+    """
+    rtree = LibSpatialIndex.RTree(2)
+    # insert an MBR for each polygon in the RTree
+    for (i, mbr) in enumerate(minimum_bounding_rects)
+        LibSpatialIndex.insert!(rtree, i, mbr[1], mbr[2])
```
Comment (matthewkol186) [inline src/graph.jl:49]: `mbr[1]` is the coordinate of the lower left corner, `mbr[2]` is the coordinate of the upper right corner

File: Project.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ version = "0.1.0"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 LibGEOS = "a90b1aa1-3769-5649-ba7e-abc5a9d163eb"
+LibSpatialIndex = "f19c2e90-9d16-5f2d-a2a7-af3fb29e4907"
```
Comment (pjrule) [inline Project.toml:10]: Question: Did you have to go through any hoops to get LibSpatialIndex installed? Totally OK with using it—indeed, I believe I may have recommended it—but I've had trouble with it in the past in a Python context.

File: src/graph.jl (hunk 1 above)
Comment (bsuwal) [reply src/graph.jl:49]: this should be a comment in the code

File: src/graph.jl [hunk 3]
```diff
@@ -39,6 +39,35 @@ function min_bounding_rect(coords::Vector{Vector{Vector{Vector{Float64}}}})::Tup
 end
 
 
+function create_rtree(minimum_bounding_rects::Array{Tuple{Array{Float64,1},Array{Float64,1}},1})::LibSpatialIndex.RTree
+    """ Given an array of minimum bounding rectangles, constructs an R-Tree
+        that will make it easier to identify candidates for intersecting nodes.
```
Comment (bsuwal) [inline src/graph.jl:44]: Could you explain the `minimum_bounding_rects` argument in more detail?

File: Project.toml (hunk 2 above)
Comment (matthewkol186) [reply Project.toml:10]: Nope, not at all! Everything was pretty smooth for me.

File: src/graph.jl (hunk 3 above)
Comment (matthewkol186) [reply src/graph.jl:44]: Yes! will add to this docstring 😄

File: src/graph.jl (hunk 1 above)
Comment (matthewkol186) [reply src/graph.jl:49]: done!

**Pull request merged**